### PR TITLE
feat: LEAP-58: Proxy URL for Taxonomy if possible

### PR DESCRIPTION
**Pull request by robot-ci-heartex**
Hi @hlomzik!

This PR was [created](https://github.com/HumanSignal/label-studio/actions/runs/6553296978) in a response to PRs in upstream repos:
- https://github.com/HumanSignal/dm2/pull/251
- https://github.com/HumanSignal/label-studio-frontend/pull/1592

### CODE REVIEW

**Review by hlomzik — APPROVED** [review]

File: label_studio/data_import/api.py [hunk 1]
```diff
@@ -712,3 +696,51 @@ def get(self, request, *args, **kwargs):
...
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """Get the presigned url for a given fileuri"""
+        request = self.request
+        task_id = kwargs.get('task_id')
+        fileuri = request.GET.get('fileuri')
+
+        if fileuri is None or task_id is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (dredivaris) [inline label_studio/data_import/api.py]: Not sure 404 is the right status code for a request with missing data.  A more appropriate response code would be 400 bad request (generally due to client error - in this case missing params)

File: label_studio/data_import/api.py [hunk 2]
```diff
@@ -712,3 +696,51 @@ def get(self, request, *args, **kwargs):
...
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """Get the presigned url for a given fileuri"""
+        request = self.request
+        project_id = kwargs.get('project_id')
+        fileuri = request.GET.get('fileuri')
+
+        if fileuri is None or project_id is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (dredivaris) [inline label_studio/data_import/api.py]: Same issue as above comment

**Review by dredivaris — DISMISSED** [review]
LGTM except for 2 response code change suggestions

**Review by bmartel — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
To enable Auto Merge for this PR also merge those PRs:
- https://github.com/HumanSignal/label-studio-frontend/pull/1592

**Pull request merged**